static/src/javascripts/projects/common/modules/ui/bannerPicker.js

```diff
@@ -61,28 +63,40 @@ const init = (banners: Array<Banner>): Promise<void> => {
                 }
             };
 
-            let hasTimedOut = false;
-
-            // checks that take longer than TIME_LIMIT are forced to fail
-            const timeout = setTimeout(() => {
-                hasTimedOut = true;
+            const hasUserAcknowledgedBanner = (): boolean =>
```

Comment:
I'd rather have this run on a per-banner setting **unless** bannerpicker can also eventually handle setting this value. Otherwise, it feels like the code is split across two separate libraries